spec/views/devise/passwords/new.html.slim_spec.rb

```diff
@@ -41,4 +41,12 @@
 
     expect(rendered).to have_link(t('links.cancel'), href: @decorated_session.cancel_link_url)
   end
+
+  it 'has sp alert for certain service providers' do
+    @sp.friendly_name = ServiceProviderSessionDecorator::SP_ALERTS.keys.first
+
+    render
+
+    expect(rendered).to have_selector('.alert')
```

Comment:
How about adding the negative of this expectation to the previous spec to verify that if the SP's name is not part of SP_ALERTS, an alert won't show up?

Reply:
Same for the other view specs below.

Reply:
👍